MekHQ/src/mekhq/gui/dialog/PartsStoreDialog.java

```diff
@@ -377,181 +386,381 @@ public void filterParts() {
         public final static int N_COL          = 9;
 
-		public PartsTableModel(ArrayList<Part> inventory) {
-			data = inventory;
-		}
+        public class TargetProxy implements Comparable<TargetProxy> {
+            private TargetRoll target;
+            private String details;
+            private String description;
+
+            public TargetProxy(@Nullable TargetRoll t) {
+                target = t;
+            }
+
+            public TargetRoll getTargetRoll() {
+                return target;
+            }
+
+            @Nullable
+            public String getDescription() {
+                if (null == target) {
+                    return null;
+                }
+                if (null == description) {
+                    description = target.getDesc();
+                }
+                return description;
+            }
+
+            @Override
+            public String toString() {
+                if (null == target) {
+                    return "-";
+                }
+
+                if (null == details) {
+                    details = target.getValueAsString();
+                    if (target.getValue() != TargetRoll.IMPOSSIBLE &&
+                        target.getValue() != TargetRoll.AUTOMATIC_SUCCESS &&
+                        target.getValue() != TargetRoll.AUTOMATIC_FAIL) {
+                        details += "+";
+                    }
+                }
+
+                return details;
+            }
+
+            private int coerceTargetRoll() {
+                int r = target.getValue();
+                if (r == TargetRoll.IMPOSSIBLE) {
+                    return Integer.MAX_VALUE;
+                }
+                else if(r == TargetRoll.AUTOMATIC_FAIL) {
+                    return Integer.MAX_VALUE-1;
+                }
+                else if(r == TargetRoll.AUTOMATIC_SUCCESS) {
+                    return Integer.MIN_VALUE;
+                }
+                return r;
+            }
+
+            @Override
+            public int compareTo(TargetProxy o) {
+                return Integer.compare(coerceTargetRoll(), o.coerceTargetRoll());
+            }
+        }
+
+        public class FormattedValue<T extends Comparable<T>> implements Comparable<FormattedValue<T>> {
+            private T value;
+            private String formatted;
+
+            public FormattedValue(T v, String f) {
+                value = v;
+                formatted = f;
+            }
+
+            public T getValue() {
+                return value;
+            }
+
+            @Override
+            public String toString() {
+                return formatted;
+            }
+
+            @Override
+            public int compareTo(FormattedValue<T> o) {
+                if (null == o) {
+                    return -1;
+                }
+                return getValue().compareTo(o.getValue());
+            }
+        }
+
+        public class PartProxy {
```

Comment:
This is pretty good, could you add a comment explaining a bit about what the purpose of it is? Same with TargetProxy.

Reply:
Yep.